python_modules/dagster/dagster_tests/core_tests/definitions_tests/test_repository_definition.py

```diff
@@ -644,3 +643,237 @@ def my_repo():
         AssetKey(["asset2"]),
     }
+
+
+def test_repo_with_resources():
+    @resource
+    def the_resource():
+        pass
+
+    @repository
+    def the_repo():
+        return [{"foo": the_resource}]
+
+    with pytest.raises(
+        CheckError,
+        match="Provided multiple resource dictionaries to repository, please provide only one.",
+    ):
+
+        @repository
+        def the_repo_multiple_resource_dicts():
+            return [{"foo": the_resource}, {"foo": the_resource}]
+
+
+def test_graph_default_resources():
+    @resource
+    def the_resource():
+        return "hello"
+
+    @op(required_resource_keys={"foo"})
+    def the_op(context):
+        assert context.resources.foo == "hello"
+
+    @graph
+    def the_graph():
+        the_op()
+
+    @repository
+    def the_repo():
+        return [{"foo": the_resource}, the_graph]
+
+    jobs = the_repo.get_all_jobs()
+    coerced_job = jobs[0]
+    assert coerced_job.name == "the_graph"
+    result = coerced_job.execute_in_process()
+    assert result.success
+
+    with pytest.raises(
+        DagsterInvalidDefinitionError,
+        match="resource key 'foo' is required by op 'the_op', but is not provided.",
+    ):
+
+        @repository
+        def the_repo_doesnt_satisfy_resource_reqs():
+            return [{"bar": the_resource}, the_graph]
+
+
+def test_job_partial_resource_spec():
+    @op(required_resource_keys={"foo", "bar"})
+    def the_op(context):
+        assert context.resources.foo == "hello"
+        assert context.resources.bar == "goodbye"
+
+    @graph
+    def the_graph():
+        the_op()
+
+    the_job = the_graph.to_partial_job(
+        resource_defs={"foo": ResourceDefinition.hardcoded_resource("hello")}
+    )
+
+    @repository
+    def the_repo():
+        return [{"bar": ResourceDefinition.hardcoded_resource("goodbye")}, the_job]
+
+    job_with_defaults_applied = the_repo.get_all_jobs()[0]
+    result = job_with_defaults_applied.execute_in_process()
+    assert result.success
+
+    with pytest.raises(
+        DagsterInvalidDefinitionError,
+        match="resource key 'bar' is required by op 'the_op', but is not provided.",
+    ):
+
+        @repository
+        def the_repo_doesnt_satisfy_resource_reqs():
+            return [{"baz": ResourceDefinition.hardcoded_resource("blah")}, the_job]
+
+
+def test_sensor_graph_coercion_dupe_names():
+    # Demonstrate that sensor coercion is broken when non-equal job with same name exists in repo
+    @op
+    def an_op():
+        pass
+
+    @op
+    def a_diff_op():
+        pass
+
+    @graph
+    def the_name():
+        an_op()
+
+    @job(name="the_name")
+    def the_job_with_name():
+        a_diff_op()
+
+    @sensor(job=the_name)
+    def the_sensor():
+        pass
+
+    @repository
+    def the_repo():
+        return [the_job_with_name, the_sensor]
+
+    assert len(the_repo.get_all_jobs()) == 1
+    assert the_repo.get_all_jobs()[0] != the_job_with_name
```

Comment:
We are probably aware of this, and it's probably not high impact, but thought I would flag.